simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -828,32 +831,44 @@ auto HdMapUtils::getFollowingLanelets(
   const lanelet::Id lanelet_id, const lanelet::Ids & candidate_lanelet_ids, const double distance,
   const bool include_self) const -> lanelet::Ids
 {
+  lanelet::Ids ids;
+  lanelet::Id previous_id;
+
   if (candidate_lanelet_ids.empty()) {
-    return {};
+    return ids;
   }
-  lanelet::Ids ids;
+
+  bool isNextLanelet{false};
   double total_distance = 0.0;
-  bool found = false;
-  for (const auto id : candidate_lanelet_ids) {
-    if (found) {
-      ids.emplace_back(id);
-      total_distance = total_distance + getLaneletLength(id);
-      if (total_distance > distance) {
-        return ids;
-      }
-    }
-    if (id == lanelet_id) {
-      found = true;
-      if (include_self) {
-        ids.push_back(id);
-      }
-    }
-  }
-  if (!found) {
+  auto lanelet = std::find_if(
+    candidate_lanelet_ids.begin(), candidate_lanelet_ids.end(),
+    [lanelet_id](const auto id) { return lanelet_id == id; });
```

Comment:
`auto lanelet_it = std::find(candidate_lanelet_ids.cbegin(), candidate_lanelet_ids.cend(), lanelet_id);`
is std::find enough here?

Reply:
done